skrf/network.py

```diff
@@ -2771,14 +2771,15 @@ def delay(self, d, unit='deg', port=0, media=None,**kw):
         if d ==0:
             return self
         d=d/2.
-        if self.nports >2:
-            raise NotImplementedError('only implemented for 1 and 2 ports')
         if media is None:
             from .media import Freespace
             media = Freespace(frequency=self.frequency,z0=self.z0[:,port])
 
         l =media.line(d=d, unit=unit,**kw)
-        return l**self
+        if port < 1:
```

Comment:
Just wondering: do you have any good reason to use < 1 instead of == 0?
```suggestion
        if port == 0:
```

Reply:
I changed to ==1